examples/targeting/add_customer_negative_criteria.rb

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright:: Copyright 2017, Google Inc. All Rights Reserved.
+#
+# License:: Licensed under the Apache License, Version 2.0 (the "License");
+#           you may not use this file except in compliance with the License.
+#           You may obtain a copy of the License at
+#
+#           http://www.apache.org/licenses/LICENSE-2.0
+#
+#           Unless required by applicable law or agreed to in writing, software
+#           distributed under the License is distributed on an "AS IS" BASIS,
+#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+#           implied.
+#           See the License for the specific language governing permissions and
+#           limitations under the License.
+#
+# This example adds various types of negative criteria to a customer. These
+# criteria will be applied to all campaigns for the customer.
+#
+#
+require 'optparse'
+require 'google/ads/google_ads'
+
+def add_customer_negative_criteria(customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  tradgedy_criterion = client.resource(:CustomerNegativeCriterion)
+  tradgedy_criterion.content_label = client.resource(:ContentLabelInfo)
+  tradgedy_criterion.content_label.type = :TRAGEDY
+
+  placement_criterion = client.resource(:CustomerNegativeCriterion)
+  placement_criterion.placement = client.resource(:PlacementInfo)
+  placement_criterion.placement.url = client.wrapper.string("http://example.com")
+
+  ops = [
+    {create: tradgedy_criterion},
+    {create: placement_criterion},
+  ]
+
+  customer_negative_criterion_service = client.service(
+    :CustomerNegativeCriterion
+  )
+
+  response = customer_negative_criterion_service.mutate_customer_negative_criteria(customer_id, ops)
+
+  if response.partial_failure_error != nil
+    raise "Got a partial failure #{response.partial_failure_error.inspect} while creating criterea"
+  end
+
+  # Display the results.
+  response.results.each do |negative_criterion|
+    puts "Customer negative criterion with criterion resource name #{negative_criterion.resource_name} was added"
+  end
+end
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf('Usage: add_campaigns.rb [options]')
```

Comment:
This isn't add_campaign.rb. The other examples grab the filename here, e.g.: https://github.com/googleads/google-ads-ruby/blob/master/examples/basic_operations/remove_ad_group.rb#L53

Reply:
👍